### DISCUSSION TITLE
Uncaught TypeError: Cannot read property 'target' of undefined

### DISCUSSION THREAD

**Discussion by f-liva** [livewire/livewire#3246] [7. Bugs (archived)] [answered]
Using this markup

```html
<select wire:change="doSomething($event.target.value)">
    <option>Foo</option>
    <option>Bar</option>
</select>
```

From the development console I get this error 

```
Uncaught TypeError: Cannot read property 'target' of undefined
    at eval (eval at parseOutMethodAndParams (wire-directives.js:93), <anonymous>:6:23)
    at Directive.parseOutMethodAndParams (wire-directives.js:93)
    at Directive.get (wire-directives.js:58)
    at LoadingStates.js:107
    at Array.map (<anonymous>)
    at processLoadingDirective (LoadingStates.js:107)
    at LoadingStates.js:22
    at Array.forEach (<anonymous>)
    at LoadingStates.js:21
    at MessageBus.js:17
```

I have to change `wire:change="doSomething($event.target.value)"` with `wire:change="doSomething(typeof $event !== 'undefined' ? $event.target.value : null)"` to workaround the issue.

Why this happens?

**Comment by joshhanley** [comment]
@f-liva if you can I would not use `wire:change` as you can just bind the select to a property instead like `wire:model="mySelection"` and it will auto update the property.

But if you do need it, `$event` will work in Livewire as it's an Alpine helper. Maybe just try `event.target.value`

Hope this helps!

**Comment by f-liva** [comment]
Utilizzando `event.target.value` il problema non sussiste più, però mi chiedo allora perché le documentazioni di Livewire riportano informazioni errate. Qui dice di usare `$event`

https://laravel-livewire.com/docs/2.x/actions#magic-actions

**Comment by joshhanley** [comment, marked as answer]
@f-liva sorry, you were right, Livewire does have an `$event` magic helper. It was broken in v2.5.2.

But a fix has been tagged for this, please upgrade to Livewire 2.5.5 and let us know if you are still having any issues 🙂

**Comment by f-liva** [comment]
Great, I will test soon thank you

Il mar 13 lug 2021, 13:00 Josh ***@***.***> ha scritto:
